metrics/ci_integrations/lib/destination/firestore/adapter/firestore_destination_client_adapter.dart

```diff
@@ -83,12 +83,107 @@ class FirestoreDestinationClientAdapter
   Future<List<BuildData>> fetchBuildsWithStatus(
     String projectId,
     BuildStatus status,
-  ) {
-    return Future.value(const []);
+  ) async {
+    logger.info(
+      'Fetching builds for the project id $projectId with status: $status...',
+    );
+
+    ArgumentError.checkNotNull(projectId, 'projectId');
+    ArgumentError.checkNotNull(status, 'status');
+
+    if (!await _projectExists(projectId)) {
+      throw ArgumentError(
+        'Project with the given ID $projectId is not found',
+      );
+    }
+
+    try {
+      final documents = await _firestore
+          .collection('build')
+          .where('projectId', isEqualTo: projectId)
+          .where('buildStatus', isEqualTo: '$status')
+          .getDocuments();
+
+      return documents.map((document) {
+        return BuildDataDeserializer.fromJson(document.map, document.id);
+      }).toList();
+    } on fd.FirestoreException catch (e) {
+      throw DestinationError(message: '$e');
+    }
   }
 
   @override
-  Future<void> updateBuilds(String projectId, List<BuildData> builds) async {}
+  Future<void> updateBuilds(String projectId, List<BuildData> builds) async {
+    logger.info(
+      'Updating builds for the project id $projectId...',
+    );
+
+    ArgumentError.checkNotNull(projectId, 'projectId');
+    ArgumentError.checkNotNull(builds, 'builds');
+
+    if (!await _projectExists(projectId)) {
+      throw ArgumentError(
+        'Project with the given ID $projectId is not found',
+      );
+    }
+
+    final updatedBuilds = builds
+        .map(
+          (build) => build.copyWith(projectId: projectId),
+        )
+        .toList();
+
+    await _updateBuilds(updatedBuilds);
+  }
+
+  /// Updates the builds having the same [BuildData.id] with the given [builds].
+  ///
+  /// Logs any errors occurred when updating builds.
+  Future<void> _updateBuilds(List<BuildData> builds) {
+    final updateFutures = <Future<void>>[];
+
+    for (final build in builds) {
+      final buildId = build.id;
+
+      final oldBuild = _firestore.collection('build').document(buildId);
+
+      final updateFuture = oldBuild.update(build.toJson()).catchError(
+        (error) {
+          logger.info(
+            'Failed to update the following build ${build.id} with the following error: $error',
+          );
+        },
+      );
```

Comment:
```suggestion
      final buildData = build.toJson();
      final reference = _firestore.document('build/${build.id}');

      final updateFuture = reference.update(buildData).catchError(_handeBuildUpdateError);
```
Does the `_handeBuildUpdateError` make sense? It seems like it simplifies this method.